File: 02/neural.m
%% 
clear all
n = 4;
x = importdata('data_x.txt');
y = importdata('data_y.txt');
dim = length(y);
%%
cvx_begin
    variable wi(n)
    minimize( norm((1+exp((1+exp(-x)).^(-1)*wi)).^(-1) -y,2 ))
cvx_end

%%
wi = randn(10,1);
norm(sigmf([zeros(dim,8) sigmf([sigmf(x,[1 0]) zeros(dim,6)]*wi,[1 0]) zeros(dim,1)] * wi + [zeros(dim,9) sigmf([zeros(dim,4) sigmf(x,[1 0]) zeros(dim,2)]*wi,[1 0])] * wi,[1 0]) - y, 2 )

%%
wi = randn(10,1);
norm(sigmf(sigmf(sigmf(x,[1,0])*([eye(4) zeros(4,6)]*wi),[1,0])*([zeros(1,8) 1 0]*wi) + sigmf(sigmf(x,[1,0])*([zeros(4,4) eye(4) zeros(4,2)]*wi),[1,0])*([zeros(1,9) 1]*wi),[1,0])- y, 2 )

%%
wi = randn(4,1);
norm(sigmf(sigmf(x,[1,0])*wi,[1,0])-y,2)

%%
wi = randn(4,1);
norm((1+exp((1+exp(-x)).^(-1)*wi)).^(-1) -y,2 )
%%

File: 02/neural.py
from cvxpy 	import *
from cvxopt import *
import numpy
import math
# That process_data.py tool
import csv

i=1
f1 = open('data_x.txt','w')
f2 = open('data_y.txt','w')

with open('clr.dat','r') as rf:
    reader = csv.reader(rf,delimiter=' ')
    for row in reader:
        f1.write(row[0]+' '+row[1]+' '+row[2]+' '+row[3]+'\n')
        f2.write(row[6]+'\n')
        i += 1

f1.close() 
f2.close()

# Initialization of w with Random Numbers
# w={} #w is a dictionary properly indexed
# for i in range(1, 5):
# 	for j in range(5, 7):
# 		w[i,j] = numpy.random.randn(1,1)[0,0]
# for i in range(5,7):
# 	w[i,7] = numpy.random.randn(1,1)[0,0]
# print w

def sigmoid(x):
	y = 1/(1+math.e**(-x))
	return y

# Import


# Modeling the problem
n = 10; # Size of the problem
w = Variable(n)

# obj = Minimize(sum_squares(sigmoid(w)))

res = sigmoid(numpy.array([1,1]))
print res

